src/aptos_by_example/3_aptos_escrow.md

```diff
@@ -0,0 +1,291 @@
+        let store = fungible_asset::create_store(constructor_ref, metadata);
+        fungible_asset::deposit(store, asset);
+
+        // Construct the Escrow resource and move it into the new object
+        let escrow = Escrow {
+            store,
+            receiver,
+            extend_ref,
+            delete_ref,
+        };
+        move_to(obj_signer, escrow);
+
+        // Emit event for tracking
+        event::emit(EscrowCreatedEvent {
+            sender: address_of(sender),
+            receiver,
+            fa: object::object_address(&metadata),
+            amount,
+            escrow_obj: object::address_from_constructor_ref(constructor_ref)
+        })
+    }
+
+    /// Allows the `sender` to cancel the escrow and retrieve the locked assets.
+    ///
+    /// Only the original sender (owner of the object) can perform this.
+    public entry fun cancel_escrow(sender: &signer, escrow_obj: Object<Escrow>) acquires Escrow {
+        assert!(object::is_owner(escrow_obj, address_of(sender)), ENOT_SENDER);
+        let escrow_obj_addr = object::object_address(&escrow_obj);
+
+        // Take full control of the escrow resource
+        let Escrow {
+            store,
+            receiver: _,
+            extend_ref,
+            delete_ref,
+        } = move_from<Escrow>(escrow_obj_addr);
+
+        let amount = fungible_asset::balance(store);
+
+        // Withdraw the asset using the signer derived from extend_ref
+        let obj_signer = &object::generate_signer_for_extending(&extend_ref);
+        let asset = fungible_asset::withdraw(obj_signer, store, amount);
+
+        // Delete the escrow object
+        object::delete(delete_ref);
+
+        // Return the assets back to sender
+        primary_fungible_store::deposit(address_of(sender), asset);
+    }
+
+    /// Allows the intended `receiver` to accept the escrow and receive the assets.
+    ///
+    /// Only the assigned receiver address is allowed to call this.
+    public entry fun accept_escrow(receiver: &signer, escrow_obj: Object<Escrow>) acquires Escrow {
+        let escrow_obj_addr = object::object_address(&escrow_obj);
+
+        // Take full control of the escrow resource
+        let Escrow {
+            store,
+            receiver: receiver_addr,
+            extend_ref,
+            delete_ref,
+        } = move_from<Escrow>(escrow_obj_addr);
+
+        // Validate that only the receiver can accept
+        assert!(receiver_addr == address_of(receiver), ENOT_RECEIVER);
+
+        let amount = fungible_asset::balance(store);
+
+        // Withdraw the asset using signer for this object
+        let obj_signer = &object::generate_signer_for_extending(&extend_ref);
+        let asset = fungible_asset::withdraw(obj_signer, store, amount);
+
+        // Delete the escrow object
+        object::delete(delete_ref);
+
+        // Deposit the assets into the receiver's account
+        primary_fungible_store::deposit(receiver_addr, asset);
```

Comment:
I would just make all of this a single private function `fun withdraw_all(receiver: address, escrow_obj: Object<Escrow>)`, and call it from within both of these functions, since both are essentially the same